### DISCUSSION TITLE
如何使用环境变量定义的 AES_SECRET_KEY

### DISCUSSION THREAD

**Discussion by txyyh** [spencerwooo/onedrive-vercel-index#229] [Q&A]
自己摸索着把
`const AES_SECRET_KEY = 'onedrive-vercel-index`

改成了

`const AES_SECRET_KEY = process.env.AES_SECRET_KEY`

遇到以下报错
![image](https://user-images.githubusercontent.com/41893946/147855396-d83f806b-4e68-4f2b-8afe-0a609d71253b.png)

本人对编程知之甚少，实在是不知道该咋改了。。。
只能看出是数据类型不对 (?)

**Comment by spencerwooo** [comment]
Vercel 里面定义 AES_SECRET_KEY 了吗？正常情况下[用 process.env.AES_SECRET_KEY 引用环境变量没啥问题](https://nextjs.org/docs/basic-features/environment-variables)

定义了的话，有可能是因为环境变量名和这里定义的 const 变量名都是 AES_SECRET_KEY 导致的？在代码里换一个变量名称试一下：

```typescript
const secretKey = process.env.AES_SECRET_KEY
// ...
export function ... {
  const encrypted = CryptoJS.AES.encrypt(token, secretKey)
  // ...
}

**Reply by txyyh** [reply]
改了 vercel 环境变量名，报错依旧

**Reply by spencerwooo** [reply]
想起来了，Next.js 的环境变量在 client side 引用需要在变量名称前面加上 `NEXT_PUBLIC_`，比如 `NEXT_PUBLIC_AES_KEY` 这样。https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser

**Reply by txyyh** [reply]
`const AES_SECRET_KEY = process.env.NEXT_PUBLIC_AES_KEY`
这样？
报错依旧

**Reply by myl7** [reply]
这样 `const AES_SECRET_KEY = process.env.NEXT_PUBLIC_AES_SECRET_KEY` 你试试？

**Reply by txyyh** [reply]
@myl7 已经改 vercel 的环境变量名为 `AES_KEY` 了

**Comment by myl7** [comment]
可以改成 `const AES_SECRET_KEY = process.env.AES_SECRET_KEY!` 试试

**Reply by txyyh** [reply]
可以了，问一下!起什么作用

**Reply by myl7** [reply]
在 Typescript 里，把 `string | undefined` 类型 assert 成 `string` 类型

**Reply by txyyh** [reply]
[#232 (Answer in thread)](https://github.com/spencerwooo/onedrive-vercel-index/discussions/232#discussioncomment-1895099)
然后 process.env.NEXT_PUBLIC_AES_KEY!的话直接 404 这样

咱目前是放弃了，恢复到明文加密
